pkg/glanceapi/statefulset.go

```diff
@@ -326,5 +326,7 @@ func StatefulSet(
 		statefulset.Spec.Template.Spec.NodeSelector = instance.Spec.NodeSelector
 	}
 
+	// Bump the ObservedGeneration for the StatefulSet CR
+	statefulset.Status.ObservedGeneration = statefulset.Generation
```

Comment:
We should not do this. The statefulset controller is expected to do such a bump when it detects the new generation.

Reply:
Ok, this is probably the result of an issue on my end. I didn't see this parameter and I wasn't able to compare it w/ Generation, but let me revisit this part.